src/robotFactory.js

```diff
@@ -1,10 +1,87 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    coords.x = coords.x
+      ? coords.x
+      : 0;
 
-class DeliveryDrone {}
+    coords.y = coords.y
+      ? coords.y
+      : 0;
+  }
+}
+
+BaseRobot.prototype.goLeft = function(step = 1) {
+  this.coords.x -= step;
+};
+
+BaseRobot.prototype.goRight = function(step = 1) {
+  this.coords.x += step;
+};
+
+BaseRobot.prototype.goBack = function(step = 1) {
+  this.coords.y -= step;
+};
+
+BaseRobot.prototype.goForward = function(step = 1) {
+  this.coords.y += step;
+};
+
+BaseRobot.prototype.getInfo = function() {
+  // eslint-disable-next-line max-len
+  return `Robot: ${this.name}, Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+};
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords = {
+    x: 0,
+    y: 0,
+    z: 0,
+  }, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    coords.z = coords.z
+      ? coords.z
+      : 0;
```

Comment:
Same as in baseRobot.

```suggestion
    this.coords.z = coords.z || 0;
```